### ISSUE TITLE
[CVE-2022-37488]/prototype-pollution found in thenable.js

### ISSUE THREAD

**Issue by Pranayk786** [moll/js-must#78]
Prototype pollution vulnerability in moll js-must 0.13.4 via the name variable in thenable.js.

The prototype pollution vulnerability can be mitigated with several best practices described here:
https://learn.snyk.io/lessons/prototype-pollution/javascript/

**Comment by moll** [comment]
Hey,

Thanks for caring about security. Fortunately, or unfortunately depending the viewpoint, I don't agree that the "prototype pollusion" class of issues is a vulnerability in Must.js, but an issue both in common JavaScript runtimes and the latest ECMAScript standard. I've written more about this here, for example: https://github.com/moll/js-oolong/#__proto__. The support for `__proto__` is easy to turn off and should be turned off in all actual applications. This is *not* a library vulnerability. Must.js is written strictly in ECMAScript 5 (apart for support for Promises, should they exist) and **must**(.js) not make accommodations for poor defaults.

In other words, please use the following in applications to turn off `__proto__` once and for all:

```js
Object.defineProperty(Object.prototype, "__proto__", {
  value: undefined, configurable: true, writable: true
})
```

May I ask which organization reserved the CVE for you? I'd like to contact them, too. Thanks!

**Comment by moll** [comment]
Oh, and the use of `for in` in [`thenable.js`](https://github.com/moll/js-must/blob/master/lib/thenable.js) is over Must's own object with known keys, so the handling of keys named `__proto__` is not relevant.
